blocks/log.py

```diff
@@ -1,274 +1,117 @@
 """The event-based main loop of Blocks."""
-from abc import ABCMeta, abstractmethod
-from collections import defaultdict
-
-from six import add_metaclass
-try:
-    from pandas import DataFrame
-    PANDAS_AVAILABLE = True
-except ImportError:
-    PANDAS_AVAILABLE = False
-
-
-@add_metaclass(ABCMeta)
-class AbstractTrainingStatus(object):
-    """The base class for objects that carry the training status.
-
-    To support various backends (such as database tables) the
-    descendants of this class are expected to override the
-    `__getattr__`, `__setattr__`, `__hasattr__` methods.
-
-    Attributes
-    ----------
-    iterations_done : int
-        The number of iterations done.
-    epochs_done : int
-        The number of epochs done.
-    _epoch_ends : list
-        The numbers of the epochs last iterations.
-
-    .. todo::
-
-        We need some notion of an attributes property. Examples of possible
-        properties include a docstring, a priority level to be used by a
-        printing extension to decide whether a certain attribute should be
-        printed or not.
-
-    """
-    def __init__(self):
-        self.iterations_done = 0
-        self.epochs_done = 0
-        self._epoch_ends = []
-
-    @abstractmethod
-    def __iter__(self):
-        """Return iterator through the status attributes.
-
-        The iterator should yield (attribute name, attribute value) pairs.
-
-        """
-        pass
+from collections import Mapping, OrderedDict
+from numbers import Integral
+from operator import methodcaller
+
+from pymongo import ASCENDING, MongoClient
```

Comment:
How difficult would it be to retain pymongo as an optional dependency? The less the better.